qlib/TableMapper.qm

```diff
@@ -880,8 +993,15 @@ on_error table_mapper.rollback();
 
             # get return value
             hash rv = hbuf + rh;
-            # clear buffer but keep hash keys
-            map hbuf.$1 = (), hbuf.keyIterator();
+            if (exists rec_list_size) {
```

Comment:
I think we should always keep `hbuf` in hash of lists + constant values format and then if the block size is 1 (such as is set if the driver doesn't support bulk DML) then insert with
```
if (insert_block == 1)
    map stmt.execArgs($1.values() + out_args), hbuf.contextIterator();
else
    stmt.execArgs(hbuf.values() + out_args);
```
in `TableMapper::flushIntern()`

this allows for simpler logic + a workaround in one place for drivers that don't support bulk DML.  SqlUtil already supports this in the `AbstractTable::insert()` method.